prusti-viper/src/encoder/encoder.rs

```diff
@@ -202,6 +205,24 @@ impl<'v, 'tcx> Encoder<'v, 'tcx> {
         self.spec
     }
 
+    /// Returns the def_id of the element containing the specifications.
+    /// This can be different from the def_id that was passed in if the
+    /// specifications were externally declared.
+    pub fn def_id(&self, def_id: &'v ProcedureDefId) -> &'v ProcedureDefId {
```

Comment:
This method name is too generic. Anything else, e.g. `get_specification_def_id`, would be better.